### PR TITLE
feat:friendly tx-id

### LINKED ISSUE #7: Friendly transaction id
Let's create a utility to represent transaction ids(or any cryptographic hash) in a friendly manner.

For displaying and sharing purposes the IDs of any transaction should be short and URL friendly. Traditionally transaction identifiers are long 256bit hashes represented in HEX that are awful to display in a UI.
One improvement on the visualization is ~base64~ base58 encoding ~(URL friendly version)~ the hash(the bytes, not the hex string) but since it will still be quite long we can trim it to be 32bit long for example before encoding to base64. The usual problem of collisions when trimming hashes shouldn't be a concern since it's for displaying purposes only and 2^32 is a big enough surface to represent transactions without collisions, also an endpoint that resolves a transaction by its short id can return a list in case of ambiguity or be forwards compatible with longer ids(e.g. 64bit).

Example hash: `0xd7bd44af293e45e0dc7583d12c6e75492410b3d74b01fe87371dc2eea1637a57`
~Short version: `171Erw`~
Short version: `6WquqG`

### PR DESCRIPTION
**Pull request by stanly-johnson**
Resolve #7

### Changes
- Util function to generate a user friendly tx-id. Encode first 32bits of given transaction-id using base58.
- added http-client to Cargo.toml to fix build error

### CODE REVIEW

File: plugins/blockchain/src/util_txid.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+/// ```
+/// use blockchain::util_txid::get_short_tx_id;
+/// let result = get_short_tx_id("0xd7bd44af293e45e0dc7583d12c6e75492410b3d74b01fe87371dc2eea1637a57");
+/// assert_eq!("6WquqG".as_bytes().to_vec(), result);
+/// ```
+pub fn get_short_tx_id(input: &str) -> Vec<u8> {
+    let input = if input.starts_with("0x") {
+        hex::decode(&input[2..]).unwrap_or_else(|_| input.into())
+    } else {
+        input.into()
```
Comment (olanod) [inline plugins/blockchain/src/util_txid.rs:12]: I think we don't want the input to be any string(or no string at all?) preferably the function is fallible and returns a result, what happens when I pass `"foo"`(I think it panics), but more importantly what does it even mean to get the short id of `"foo"`?

**Review by olanod — CHANGES_REQUESTED** [review]
Hi! thanks for tackling this. What do you say about making this utility a bit more general purpose? It doesn't have to apply only to transaction ids(I'm not even sure yet about transactions having ids) but can be used as a way to "pretty print" binary data in general(making most sense for hashes). One option that comes to mind could be an extension trait for types that can be represented as bytes(e.g. `T: AsRef<[u8]>`), imagine a `Vec<u8>` or similar doing something like `data.short()` or `data.pretty()`. We could even get more strict and also require the  `core::hash::Hash` bound to ensure the data is random looking and we'll get a nice unique looking short ID but there might be cases we want to create the short ID for non hashes 🤔

File: plugins/blockchain/src/util_txid.rs (hunk 1 above)
Comment (stanly-johnson) [reply plugins/blockchain/src/util_txid.rs:12]: Yes you're right. I'll convert this to a trait as you said above.

File: plugins/blockchain/Cargo.toml [hunk 2]
```diff
@@ -13,10 +13,12 @@ hex = "0.4.2"
 jsonrpc = { version = "0.12.0", default-features = false }
 once_cell = "1.5.2"
 parity-scale-codec = "1.3.6"
+http-client = "6.3.0"
```
Comment (olanod) [inline plugins/blockchain/Cargo.toml:16]: I guess this one is not needed?

**Review by olanod — CHANGES_REQUESTED** [review]
Awesome! looks good. One last change, can you move this to the root maybe in a crate called `util`? this function won't actually be used in the blockchain plugin which is intended to be lower level and very general to work with any substrate chain, it will be used initially in the yet to be created `transfers` plugin or other higher level more Valiu specific plugins that the app will consume and sexy identifiers will be more handy.

File: plugins/blockchain/Cargo.toml (hunk 2 above)
Comment (stanly-johnson) [reply plugins/blockchain/Cargo.toml:16]: I'm not sure, removing this gives me a build error about `TlsStream` and `TcpStream`

Comment (olanod) [reply plugins/blockchain/Cargo.toml:16]: Ok weird, I can look into that later

File: plugins/blockchain/Cargo.toml [hunk 3]
```diff
@@ -13,10 +13,12 @@ hex = "0.4.2"
...
 path-tree = "0.1.12"
 surf = { version = "2.1.0", default-features = false, features = ["h1-client"] }
 twox-hash = { version = "1.6.0", default-features = false }
 valor = { version = "0.4.7-beta.0", package = "valor_core", features = ["util"] }
+bs58 = "0.4.0"
```
Comment (olanod) [inline plugins/blockchain/Cargo.toml:21]: But then what about this one? ✂️

**Review by olanod — APPROVED** [review]

**Pull request merged**